File: 2020/01/solution.rb
require_relative '../../lib/advent_helper'
helper = AdventHelper.new(script_root:__dir__, script_file: __FILE__)
input = helper.auto_parse

# Part 1
input.combination(2).each { |args| p args.reduce(&:*) if args.sum == 2020 }

# Part 2
input.combination(3).each { |args| p args.reduce(&:*) if args.sum == 2020 }

File: 2020/01/solution.exs
Code.require_file("helpers.exs", __DIR__ <> "/../lib")

entries = __DIR__ <> "/" <> "input.txt"
  |> File.read!()
  |> String.split("\n", trim: true)
  |> Enum.map(fn line ->
    {integer, _leftover} = Integer.parse(line)
    integer
  end)

entries
  |> Helpers.combinations(2)
  |> Enum.find(fn x -> Enum.sum(x) == 2020 end)
  |> Enum.reduce(fn x, acc -> x * acc end)
  |> IO.inspect

entries
  |> Helpers.combinations(3)
  |> Enum.find(fn x -> Enum.sum(x) == 2020 end)
  |> Enum.reduce(fn x, acc -> x * acc end)
  |> IO.inspect